### PR TITLE
Support BERTSQuAD and BERTNER

### PR DESCRIPTION
**Pull request by hkvision**
- Some enhancement and bug fixes.
- Add BERTSQuAD
- Will remove the examples afterwards.
- Optimizer based on #1336 
@jason-dai Please take a look.

### CODE REVIEW

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py [hunk 1]
```diff
@@ -199,6 +199,7 @@ def to_floats(vs):
                                               IdentityCriterion(),
                                               batch_size=batch_size,
                                               optim_method=self.optim_method)
+        self.optimizer.set_gradclip_l2norm(1.0)
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: Only added here temporarily for test. Issue #1281 for discussion.

**Comment by hkvision** [comment]
@yangw1234 Take a look at the changes to your code?

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py [hunk 2]
```diff
@@ -224,8 +224,9 @@ def _get_arguments_from_loss(loss, optim_method, session, val_outputs, val_label
         variables = []
         grads = []
         for (grad, var) in grads_vars:
-            variables.append(var)
-            grads.append(grad)
+            if grad is not None:
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: synced with @yangw1234 for this change offline.

**Comment by yangw1234** [comment]
estimator part looks good to me

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.211:18888/job/ZOO-PR-Validation/2357/
Merge this first as the code becomes messy. Feel free to raise further comments if any.

**Pull request merged**